PayItForward.ConsoleClient.UnitTests/Web/Controllers/StoriesController/Donate_Should.cs

```diff
@@ -0,0 +1,251 @@
+﻿namespace PayItForward.UnitTests.Web.Controllers.StoriesController
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Security.Claims;
+    using AutoMapper;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Mvc;
+    using Moq;
+    using PayItForward.Models;
+    using PayItForward.Services.Abstraction;
+    using PayItForward.Web.Models.DonationViewModels;
+    using Xunit;
+
+    public class Donate_Should
+    {
+        private readonly PayItForward.Web.Controllers.StoriesController storiesController;
+        private readonly Mock<IStoriesService> storiesService;
+        private readonly Mock<IUsersService> usersService;
+        private readonly Mock<IDonationsService> donationService;
+        private readonly Mock<IMapper> mapper;
+        private readonly Mock<IHttpContextAccessor> httpaccessor;
+        private Guid storyId = new Guid("E6E635AB-6AD9-40BD-9992-08D5D82FC3F0");
+
+        public Donate_Should()
+        {
+            this.mapper = new Mock<IMapper>();
+            this.storiesService = new Mock<IStoriesService>();
+            this.donationService = new Mock<IDonationsService>();
+            this.usersService = new Mock<IUsersService>();
+            this.httpaccessor = new Mock<IHttpContextAccessor>();
+            this.storiesController = new PayItForward.Web.Controllers.StoriesController(
+                this.storiesService.Object,
+                this.donationService.Object,
+                this.usersService.Object,
+                this.mapper.Object,
+                this.httpaccessor.Object);
+        }
+
+        [Fact]
+        public void ReturnARedirectToIndexHomeWhenIdIsEmpty()
+        {
+            // Arrange
+
+            // Act
+            var result = this.storiesController.Donate(id: Guid.Empty);
+
+            // Assert
+            var redirectToActionResult = Assert.IsType<RedirectToActionResult>(result);
+            Assert.Equal("Stories", redirectToActionResult.ControllerName);
+            Assert.Equal("Details", redirectToActionResult.ActionName);
+        }
+
+        [Fact]
+        public void ReturnContent_WithStoryNotFound_WhenStoryNotFound()
+        {
+            // Arrange
+            this.storiesService.Setup(s => s.GetStoryById(this.storyId))
+                .Returns((StoryDTO)null);
+
+            // Act
+            var result = this.storiesController.Donate(this.storyId);
+
+            // Assert
+            var contentResult = Assert.IsType<ContentResult>(result);
+            Assert.Equal("Story not found.", contentResult.Content);
+        }
+
+        [Fact]
+
+        public void ReturnResultContent_WithNoUserToDonate_WhenDonatorIdIsNullOrEmpty()
+        {
+            // Arrange
+            this.storiesService.Setup(s => s.GetStoryById(this.storyId))
+                .Returns(this.HelperStoryDto().FirstOrDefault());
```

Comment:
HelperStoryDto can be renamed to GetTestStoriesDto because it is method and it should talk about what it is